Medal splash

DrabWeb opened the pull request:
![screen shot 2017-07-11 at 10 17 22 pm](https://user-images.githubusercontent.com/6215387/28097740-6347cb48-6688-11e7-8328-8c0e5a2ce51f.png)

Still WIP because the `AsyncLoadWrapper` around the `MedalSprite` doesn't autosize and I can't figure it out at all.

DrabWeb committed:
Initial layout and animation

DrabWeb committed:
Merge https://github.com/ppy/osu into medal-splash

DrabWeb committed:
Proper dismissing.

DrabWeb committed:
Merge https://github.com/ppy/osu into medal-splash

DrabWeb committed:
Particles.

DrabWeb committed:
Make the background strip expand from the disc edges.

DrabWeb committed:
Medal sprite, make MedalOverlay auto-show when loaded.

DrabWeb committed:
Cleanup.

DrabWeb committed:
Merge https://github.com/ppy/osu into medal-splash

DrabWeb committed:
Update with framework changes.

DrabWeb committed:
Async medal sprite loading.

DrabWeb committed:
Cleanup.

DrabWeb committed:
CI fixes.

DrabWeb committed:
Convert MedalOverlay animation durations to constants.

turbedi commented on osu.Game/Overlays/MedalOverlay.cs:
```diff
@@ -0,0 +1,301 @@
+        private readonly Container backgroundStrip, particleContainer;
+        private readonly BackgroundStrip leftStrip, rightStrip;
+        private readonly CircularContainer disc;
+        private readonly Sprite innerSpin, outterSpin;
```
Typo, should be `outerSpin`

DrabWeb committed:
Better medal sprite loading, fade in particles, visual test update.

DrabWeb committed:
Typo.

peppy commented on osu.Game/Overlays/MedalOverlay.cs:
```diff
@@ -0,0 +1,304 @@
+                Anchor = Anchor.TopCentre,
+                Origin = Anchor.TopCentre,
+                RelativeSizeAxes = Axes.X,
+            }, m => { disc.Add(m); Show(); });
```
newlines between statements please

DrabWeb committed:
Formatting.

peppy committed:
Merge remote-tracking branch 'upstream/master' into DrabWeb/medal-splash

peppy committed:
Nest delays and implement IStateful, allowing for flushing on early dismiss

Note that this will break rotation loops until https://github.com/ppy/osu-framework/issues/900 is addressed.

peppy committed:
Remove unnecessary alpha/alwayspresent changes

peppy committed:
Move async loading to a higher level to simplify logic

peppy committed:
Formatting
